src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +74,651 @@ public ByteArraySet createSmallSet() {
+        }
+        assertEquals(size, set.size());
+
+        Iterator<ByteArray> iter = set.iterator();
+        List<ByteArray> alreadySeenVals = new ArrayList<>();
+        int count = 0;
+        while(iter.hasNext()) {
+            ByteArray val = iter.next();
+            assertFalse(alreadySeenVals.contains(val));
+            alreadySeenVals.add(val);
+            assertTrue(vals.contains(val));
+            count++;
+        }
+        assertEquals(size, count);
+    }
+
+    @Test
+    public void emptyIterator() {
+        testIterator(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallIterator() {
+        testIterator(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallIterator() {
+        testIterator(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularIterator() {
+        testIterator(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigIterator() {
+        testIterator(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigIterator() {
+        testIterator(SIZE_REALLY_BIG);
+    }
+
+    private void testRemove(int size) {
+        System.out.println("Testing remove method with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+        Collections.shuffle(vals);
+
+        int count = 0;
+        for(ByteArray val: set) {
+            assertTrue(vals.contains(val));
+            count++;
+        }
+        assertEquals(size, count);
+
+        int currentSize = size;
+        for(ByteArray val: vals) {
+            assertTrue(set.remove(val));
+            assertFalse(set.contains(val));
+            currentSize -= 1;
+            assertEquals(currentSize, set.size());
+        }
+    }
+
+    @Test
+    public void emptyRemove() {
+        testRemove(SIZE_EMPTY);
+    }
+
+    @Test
+    public void reallySmallRemove() {
+        testRemove(SIZE_REALLY_SMALL);
+    }
+
+    @Test
+    public void smallRemove() {
+        testRemove(SIZE_SMALL);
+    }
+
+    @Test
+    public void regularRemove() {
+        testRemove(SIZE_REGULAR);
+    }
+
+    @Test
+    public void bigRemove() {
+        testRemove(SIZE_BIG);
+    }
+
+    @Test
+    public void reallyBigRemove() {
+        testRemove(SIZE_REALLY_BIG);
+    }
+
+    private void testRandomSerializeDerializeSizeCheckRemove(int size) {
+        System.out.println("Testing serialize/deserialize methods with random-size check with " + size + " size");
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> originalVals = getRandomByteArrays(size);
+
+        List<ByteArray> vals = new ArrayList<>();
+        for (ByteArray val: originalVals) {
+            vals.add(val);
+        }
+        Collections.shuffle(vals);
```

Comment:
```suggestion
List<ByteArray> originalVals = getRandomByteArrays(size);
List<ByteArray> vals = new ArrayList<>(originalVals);
Collections.shuffle(vals);
```

you might actually want shuffle here, but this looks like it can be simplified with